### PR TITLE
separate notebook info store from editor registration

### PR DESCRIPTION
**Pull request by amunger**
refactor: remove the externalEditor flag on notebook info and just register editors independently of adding notebook info.

### CODE REVIEW

File: src/vs/workbench/contrib/notebook/browser/services/notebookServiceImpl.ts [hunk 1]
```diff
@@ -640,13 +640,12 @@ export class NotebookService extends Disposable implements INotebookService {
...
 			priority: RegisteredEditorPriority.default,
-			selectors: [],
-			externalEditor: !!data.externalEditor
+			selectors: []
 		});
 
 		info.update({ selectors: data.filenamePattern });
 
-		const reg = this.notebookProviderInfoStore.add(info);
+		const reg = this.notebookProviderInfoStore.addNotebookInfo(info);
```
Comment (rebornix) [inline src/vs/workbench/contrib/notebook/browser/services/notebookServiceImpl.ts:648]: We still want to call `this.notebookProviderInfoStore.registerEditorForNotebook` in `mainThreadNotebook#$registerNotebookSerializer`, right?

File: src/vs/workbench/contrib/notebook/browser/services/notebookServiceImpl.ts [hunk 2]
```diff
@@ -101,7 +105,7 @@ export class NotebookProviderInfoStore extends Disposable {
 
 		const builtinProvidersFromCache: Map<string, IDisposable> = new Map();
 		builtins.forEach(builtin => {
-			builtinProvidersFromCache.set(builtin.id, this.add(builtin));
+			builtinProvidersFromCache.set(builtin.id, this.addNotebookInfo(builtin));
```
Comment (rebornix) [inline src/vs/workbench/contrib/notebook/browser/services/notebookServiceImpl.ts:108]: maybe we add some comments there mentioning that builtin providers should register `Editor` themselves.

File: src/vs/workbench/contrib/notebook/browser/services/notebookServiceImpl.ts (hunk 1 above)
Comment (rebornix) [reply src/vs/workbench/contrib/notebook/browser/services/notebookServiceImpl.ts:648]: For example, Live Share registers serializers through code only, which is only calling `$registerNotebookSerializer`

**Review by rebornix — CHANGES_REQUESTED** [review]

File: src/vs/workbench/contrib/notebook/browser/services/notebookServiceImpl.ts (hunk 1 above)
Comment (amunger) [reply src/vs/workbench/contrib/notebook/browser/services/notebookServiceImpl.ts:648]: ah, I didn't notice that special case. An alternative is just to check the notebook provider is built-in every time and don't register a notebook editor if so. https://github.com/microsoft/vscode/pull/187070

We already are doing this when restoring info from the momento, so why not just make it a general check?

**Pull request closed without merging**